python/cudf/cudf/bindings/cudf_cpp.pyx

```diff
@@ -163,8 +268,16 @@ cdef set_scalar_value(gdf_scalar *scalar, val):
         scalar.data.si08 = val
     elif val.dtype.type == np.bool or val.dtype.type == np.bool_:
         scalar.data.b08 = val
-    elif val.dtype.name == 'datetime64[ms]':
-        scalar.data.dt64 = np.datetime64(val, 'ms').astype(int)
+    elif val.dtype.type == np.datetime64:
+        time_unit, _ = np.datetime_data(val.dtype)
+        if time_unit == 'ms':
+            scalar.data.dt64 = np.datetime64(val, time_unit).astype(int)
+        elif time_unit in np_to_gdf_time_unit:
+            scalar.data.tmst = np.datetime64(val, time_unit).astype(int)
+        elif time_unit == 'D':
+            scalar.data.dt32 = np.datetime64(val, time_unit).astype(int)
```

Comment:
Shouldn't do this given the sizing concerns, lets just not implement DATE32 yet.

Reply:
Yeah, I went ahead and removed all references to DATE32 in the latest changeset.